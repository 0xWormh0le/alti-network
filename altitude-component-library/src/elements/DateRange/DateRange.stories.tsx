import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import DateRange, { DateRangeProps } from './DateRange'

export default {
  title: 'Elements/DateRange',
  component: DateRange
} as Meta

const Template: Story<DateRangeProps> = (args) => <DateRange {...args} />

export const Default = Template.bind({})
Default.args = {
  from: 1,
  to: 2312312
}
