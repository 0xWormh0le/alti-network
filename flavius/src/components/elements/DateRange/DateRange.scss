.DateRange {
  &__date {
    display: inline-block;
  }
}
