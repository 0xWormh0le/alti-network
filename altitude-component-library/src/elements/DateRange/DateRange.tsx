import React from 'react'
import Moment from 'react-moment'
import './DateRange.scss'

export interface DateRangeProps {
  from?: number
  to?: number
  className?: string
  displayFormat: string
}

export const DateRange: React.FC<DateRangeProps> = ({ from, to, className, displayFormat }) => {
  const formattedFrom = (
    <Moment unix={true} format={displayFormat} className='DateRange__date'>
      {from}
    </Moment>
  )
  const formattedTo = (
    <Moment unix={true} format={displayFormat} className='DateRange__date'>
      {to}
    </Moment>
  )

  return (
    <div className={`DateRange ${className}`}>
      {formattedFrom} - {formattedTo}
    </div>
  )
}

export default DateRange
