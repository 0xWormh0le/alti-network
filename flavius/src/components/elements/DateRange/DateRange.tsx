import React from 'react'
import Moment from 'react-moment'
import CONSTANTS from 'util/constants'
import './DateRange.scss'

const { TIME_DISPLAY_FORMAT } = CONSTANTS

export interface DateRangeProps {
  from?: number
  to?: number
  className?: string
}

export const DateRange: React.FC<DateRangeProps> = ({ from, to, className }) => {
  const formattedFrom = (
    <Moment unix={true} format={TIME_DISPLAY_FORMAT.DATE_FORMAT} className='DateRange__date'>
      {from}
    </Moment>
  )
  const formattedTo = (
    <Moment unix={true} format={TIME_DISPLAY_FORMAT.DATE_FORMAT} className='DateRange__date'>
      {to}
    </Moment>
  )

  return (
    <div className={`DateRange ${className}`}>
      {formattedFrom} - {formattedTo}
    </div>
  )
}

export default DateRange
