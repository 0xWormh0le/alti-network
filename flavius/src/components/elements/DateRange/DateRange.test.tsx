import React from 'react'
import { render } from '@testing-library/react'
import { DateRange, DateRangeProps } from './DateRange'

describe('DateRange', () => {
  it('renders correctly', () => {
    const props: DateRangeProps = {
      from: 1566492823,
      to: 1566502823,
      className: 'DateRange--custom'
    }
    const { container } = render(<DateRange {...props} />)
    expect(container).toMatchSnapshot()
  })
})
