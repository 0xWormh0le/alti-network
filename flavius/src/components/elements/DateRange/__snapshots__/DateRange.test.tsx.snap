// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRange renders correctly 1`] = `
<div>
  <div
    class="DateRange DateRange--custom"
  >
    <time
      class="DateRange__date"
      datetime="1566492823000"
    >
      Aug 22, 2019
    </time>
     - 
    <time
      class="DateRange__date"
      datetime="1566502823000"
    >
      Aug 22, 2019
    </time>
  </div>
</div>
`;
